### PR TITLE
Address issue about parallel transport API

### PR DESCRIPTION
**Pull request by nguigs**
Start addressing #1187

### CODE REVIEW

File: examples/geodesic_regression_hypersphere.py [hunk 1]
```diff
@@ -73,8 +73,8 @@ def main():
     tangent_vec_of_transport = METRIC.log(intercept, base_point=intercept_hat)
     transported_coef_hat = METRIC.parallel_transport(
         tangent_vec_a=tangent_vec_to_transport,
```
Comment (ninamiolane) [inline examples/geodesic_regression_hypersphere.py:75]: Should we rename tangent_vec_a and tangent_vec_b : they play different roles, and I always have to go back to the file implementing the parallel transport to remember which one is which one.

How about: `tangent_vec_to_transport` and `tangent_vec`?

File: geomstats/geometry/connection.py [hunk 2]
```diff
@@ -627,22 +627,29 @@ def path(t):
...
-        tangent_vec_b)`.
+        along the geodesic between two points `base_point` and `end_point`
+        or alternatively defined by :math:`t\mapsto exp_(base_point)(
+        t*tangent_vec_b)`.
 
         Parameters
         ----------
         tangent_vec_a : array-like, shape=[..., {dim, [n, m]}]
             Tangent vector at base point to be transported.
+        base_point : array-like, shape=[..., {dim, [n, m]}]
```
Comment (ninamiolane) [inline geomstats/geometry/connection.py:644]: how about `initial_point` to be consistent with the geodesic's API?

File: geomstats/geometry/symmetric_matrices.py [hunk 3]
```diff
@@ -228,9 +228,13 @@ def apply_func_to_eigvals(mat, function, check_positive=False):
         """
         eigvals, eigvecs = gs.linalg.eigh(mat)
         if check_positive and gs.any(gs.cast(eigvals, gs.float32) < 0.0):
-            logging.warning(
-                "Negative eigenvalue encountered in" " {}".format(function.__name__)
-            )
+            try:
```
Comment (ninamiolane) [inline geomstats/geometry/symmetric_matrices.py:231]: what is this? and does this have to be coded with a try/except? why not if and raise Error?

File: tests/helper.py [hunk 4]
```diff
@@ -96,35 +96,39 @@ def is_isometry(tan_a, trans_a, endpoint):
     tan_vec_b = space.to_tangent(gs.random.rand(*shape) / 5, base_point)
     end_point = metric.exp(tan_vec_b, base_point)
 
-    transported = metric.parallel_transport(tan_vec_a, tan_vec_b, base_point)
+    transported = metric.parallel_transport(tan_vec_a, base_point, tan_vec_b)
+    result = is_isometry(tan_vec_a, transported, end_point)
+    results.append(gs.all(result))
+
+    transported = metric.parallel_transport(tan_vec_a, base_point, end_point=end_point)
```
Comment (ninamiolane) [inline tests/helper.py:103]: do you have a test for vectorization when called with base_point/end_point?

Comment (ninamiolane) [reply tests/helper.py:103]: Could we also add a test that checks that for the same geodesic (of initial_point, end_point and initial_tangent_vec_) both versions of the parallel transport give the same result?

File: tests/tests_geomstats/test_pre_shape.py [hunk 5]
```diff
@@ -766,3 +766,9 @@ def test_parallel_transport(self):
         is_horizontal = space.is_horizontal(transported, end_point)
         self.assertTrue(gs.all(is_tangent))
         self.assertTrue(gs.all(is_horizontal))
+
+        transported = metric.parallel_transport(
+            tan_a[0], point, end_point=end_point[0], n_steps=150, step="rk4"
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_pre_shape.py:771]: could we check that both versions of the parallel transport give the same result?

**Review by ninamiolane — APPROVED** [review]
LGTM! Two main comments:
- the API of the parallel transport could benefit matching the API of the geodesic: initial_point, end_point, initial_tangent_vec, tangent_vec_to_transport?
- could you add slightly more tests, i.e. checking that both parallel transports give the same transported vector?

Otherwise good to merge! 🎉

File: examples/geodesic_regression_hypersphere.py (hunk 1 above)
Comment (nguigs) [reply examples/geodesic_regression_hypersphere.py:75]: what about `tangent_vec` and `initial_direction`?

File: geomstats/geometry/connection.py (hunk 2 above)
Comment (nguigs) [reply geomstats/geometry/connection.py:644]: Ok

File: geomstats/geometry/symmetric_matrices.py (hunk 3 above)
Comment (nguigs) [reply geomstats/geometry/symmetric_matrices.py:231]: I got an error from this line when function is a list of functions (in powerm).

File: tests/helper.py (hunk 4 above)
Comment (nguigs) [reply tests/helper.py:103]: This is with multiple inputs for all args, I will try with only one `end_point`.

File: tests/tests_geomstats/test_pre_shape.py (hunk 5 above)
Comment (nguigs) [reply tests/tests_geomstats/test_pre_shape.py:771]: What do you mean both versions? Do mean ladder methods? This check is done for the hypersphere.

Comment (ninamiolane) [reply tests/tests_geomstats/test_pre_shape.py:771]: Sorry I wasn't clear. i mean a version with initial_point and end_point, and a version with initial_point and initial_tangent_vec.

File: examples/geodesic_regression_hypersphere.py (hunk 1 above)
Comment (ninamiolane) [reply examples/geodesic_regression_hypersphere.py:75]: or tangent_vec and initial_tangent_vec? which ever you prefer!

**Pull request merged**